cmd/common.go

```diff
@@ -318,6 +325,9 @@ func syncMain(ctx context.Context, filenames []string, dry bool, parallelism,
 	if err := checkForRBACResources(*rawState, dumpConfig.RBACResourcesOnly); err != nil {
 		return err
 	}
+	if err := checkFilterChainsAllowed(*rawState, dumpConfig, parsedKongVersion); err != nil {
```

Comment:
Is this even needed? If decK attempts to create a filter chain on a Gateway that doesn't support it, can't we just return the API error back to the caller?

Right now decK is mostly best effort when it comes to features driven by Kong configuration, for example decK doesn't check whether a custom plugin exists before attempting to configure it.

Reply:
If we don't do some kind of feature detection, pretty much all deck commands will throw an error when executing against a Kong node without wasm support (too old version, not turned on, etc) because `GET /filter-chains` will return 404/405.

So you are correct that `checkFilterChainsAllowed` is not _strictly_ required, but `determineFilterChainSupport` (feature detection) is. I would prefer to keep the `checkFilterChainsAllowed` guard in place because it's nicer on the user to fail fast and protects against things like incomplete sync, but I can remove it if you think it's just not a good fit with the rest of the codebase.

Reply:
> GET /filter-chains will return 404/405.

For reference, we do this already for Consumer Groups: https://github.com/Kong/go-database-reconciler/blob/main/pkg/dump/dump.go#L90

I'm okay if you prefer to do this extra check though.

Reply:
Ahhhhh interesting. I think I prefer it that way actually. I'll do some testing and push this logic to go-database-reconciler.

Reply:
https://github.com/Kong/go-database-reconciler/pull/101